processing/src/main/java/io/druid/query/filter/SelectorDimFilter.java

```diff
@@ -88,10 +87,16 @@ public Filter toFilter()
       return new SelectorFilter(dimension, value);
     } else {
       final String valueOrNull = Strings.emptyToNull(value);
-      final Predicate<String> predicate = new Predicate<String>()
+      final DruidCompositePredicate predicate = new DruidCompositePredicate()
       {
         @Override
-        public boolean apply(String input)
+        public boolean applyLong(long value)
+        {
+          return Objects.equals(valueOrNull, String.valueOf(value));
```

Comment:
This should convert `valueOrNull` to a long and then compare directly to `long input`.